### ISSUE TITLE
Git pre-commit hook fails to run because of "No such file or directory"

### ISSUE THREAD

**Issue by saschanaz** [desktop/desktop#12586]
### Describe the bug

I have a `.git/hooks/pre-commit` which is exactly the following:

```sh
#!/bin/sh
npm test
```

When I commit via GitHub Desktop, it fails:

```
/bin/bash: C:/Program Files/nodejs/npm: No such file or directory
```

It works when committing via git CLI.

### Version & OS

Open 'About GitHub Desktop' menu to see the Desktop version. Also include what operating system you are using.

### Steps to reproduce the behavior

1. Create a hook as `.git/hooks/pre-commit`
2. Commit something

### Expected behavior

It should run the npm command

### Actual behavior

It says no such file

### Screenshots

![image](https://user-images.githubusercontent.com/3396686/125094944-f9498b80-e0d3-11eb-89fd-3990a4ccdebe.png)

### Logs

```
2021-07-09T14:34:12.623Z - info: [ui] [Timing] Action 'create commit' for 'saschanaz/types-web' took 0.292s
2021-07-09T14:35:53.954Z - info: [ui] Executing createCommit: git commit -F - (took 1.372s)
2021-07-09T14:35:53.963Z - error: [ui] `git commit -F -` exited with an unexpected code: 1.
stderr:
/bin/bash: C:/Program Files/nodejs/npm: No such file or directory
```

### Additional context

This had been working but is regressed since a few months ago.

**Comment by steveward** [comment]
@saschanaz thanks for the report. I tested this out in a Windows VM and was unable to reproduce this, so my guess is it's something specific to your environment. Can you try launching GitHub Desktop via the command line using the `github` command and see if that resolves this issue?

**Comment by saschanaz** [comment]
> Can you try launching GitHub Desktop via the command line using the `github` command and see if that resolves this issue?

No, that does not solve the issue, the same message opens when committing anything.

**Comment by ruggi99** [comment]
@saschanaz Have you installed a node version manager in your environment?
I have the same issue and I'm trying to tackle it down

**Comment by saschanaz** [comment]
No, but winget should probably work for that purpose I guess.

**Comment by ruggi99** [comment]
Nevermind.
I have the same issue, like already reported in #11603

**Comment by saschanaz** [comment]
I don't think it's the same issue if your issue has symlink involved.

**Comment by ruggi99** [comment]
Yeah could be, or could be related for some reason

**Comment by ruggi99** [comment]
Symlinks was an idea.
I tried changing `npm` to `npm.cmd` (supposing you are on Windows) and got it working finally.
Don't know why but it works. But the problem is that pre-commit hook will not be cross-platform anymore
Does it work for you too?

**Comment by tomiGoljevscek** [comment]
> I tried changing npm to npm.cmd

This worked for me, thank you so much!

**Comment by Airkro** [comment]
I think this is the same problem as #12562.

**Comment by Wangyaqi** [comment]
> Insert git bin to PATH and put it before "%SystemRoot%\system32". 

It works for me. It looks like Github Desktop used the wrong bash command which was provided by wsl.

**Comment by WellingtonOstemberg** [comment]
> 

Solved my problem!
Finally

**Comment by murugan1993** [comment]
missing Change-Id in commit message footer this my error what to do can you explain

**Comment by rdrinovec** [comment]
> > Insert git bin to PATH and put it before "%SystemRoot%\system32".
> 
> It works for me. It looks like Github Desktop used the wrong bash command which was provided by wsl.

Does not work for me 😢

**Comment by lucasnguyen3979** [comment]
> > > Insert git bin to PATH and put it before "%SystemRoot%\system32".
> > 
> > 
> > It works for me. It looks like Github Desktop used the wrong bash command which was provided by wsl.
> 
> Does not work for me 😢

At first it didn't work for me, but then I added `C:\Program Files\Git\bin` to `Path` for both variable types: **User** and **System**, and it worked.

**Comment by TheMikeyRoss** [comment]
2 years later, the same problem is happening. And updating the PATH doesn't fix it

**Comment by MeridianGH** [comment]
> 2 years later, the same problem is happening. And updating the PATH doesn't fix it

The weird thing is, I just recently set up husky, ran into this issue, got it fixed by adding `C: \Program Files\Git\bin` to `PATH` until it broke on me again just a few days ago. I obviously did not change the `PATH`.

Maybe a newer version of GitHub Desktop broke things? (I'm currently on version `3.4.0 (x64)`) It looks like the Windows `PATH` is not respected at all, as the error message I'm getting from husky looks like it's using MinGW:

```
.husky/commit-msg: line 4: npx.cmd: command not found
husky - commit-msg script failed (code 127)
husky - command not found in PATH=/mingw64/libexec/git-core:/mingw64/bin:/usr/bin:/c/Users/[my username]/bin:/mingw64/bin:/mingw64/usr/bin:undefined
```

**Comment by TheMikeyRoss** [comment]
@MeridianGH apparently something changed in Github Desktop in v3.4.0 that cause git hooks issues. It's being tracked in https://github.com/desktop/desktop/issues/18739

**Comment by MeridianGH** [comment]
> @MeridianGH apparently something changed in Github Desktop in v3.4.0 that cause git hooks issues. It's being tracked in #18739

Thanks a lot, that is definitely the problem. I will hook into that issue then.

**Comment by R-Mohammed-Hasan** [comment]
**_Github desktop formats files after committting precommit_**

I am facing an issue where precommit format command works after committing the files. I mean the files are getting formatted only after committing the files. So I need to commit 2 times it seems...

**Comment by jflemingframe** [comment]
for me C:\ProgramData\nvm was in my PATH, and messed this up.

**Comment by niik** [comment]
Yesterday we shipped GitHub Desktop 3.5.5-beta1 to our [beta channel](https://desktop.github.com/beta/) with [improved support for hooks related to committing](https://github.com/desktop/desktop/pull/21319).

<img width="280" alt="Screenshot showing the GitHub Desktop commit form as a pre-commit hook is running" src="https://github.com/user-attachments/assets/b389d45d-90d5-4660-9bab-f9b6b64c8201" />

The support is experimental and currently limited to our beta channel. While we believe that this release resolved this issue we're looking for real-world feedback so if you have the ability we'd appreciate it if you could download our [beta channel](https://desktop.github.com/beta/) and give it a spin. Ensure the experimental hooks support it's enabled by going to Settings -> Git -> Hooks and making sure the "Load Git hook environment variables from shell" is ticked.

<img width="614" height="558" alt="Screenshot showing the hook settings dialog" src="https://github.com/user-attachments/assets/4ed1157c-1c89-4733-88db-dd8fe84cf58f" />
